components/TextFields/MDCTextInput.h

```diff
@@ -0,0 +1,182 @@
+   */
+  MDCTextInputPresentationStyleDefault = 0,
+
+  /**
+   The placeholder text is laid out inline but will float above the field when there is content or
+   the field is being edited. The character count is below text. The Material Design guidelines
+   call this 'Floating inline labels.'
+   https://material.io/guidelines/components/text-fields.html#text-fields-labels
+   */
+  MDCTextInputPresentationStyleFloatingPlaceholder,
+
+  /** The placeholder is laid out inline and the character count is also inline to the right. */
+  MDCTextInputPresentationStyleFullWidth,
+};
+
+/** Common API for Material Design themed text inputs. */
+@protocol MDCTextInput <NSObject>
+
+/** A Boolean value indicating whether the text field is currently in edit mode. */
+@property(nonatomic, readonly, getter=isEditing) BOOL editing;
+
+/**
+ The text to be displayed when no text has been entered. The Material Design guidelines call this
+ 'Hint text.'
+
+ If presentationStyle == MDCTextInputPresentationStyleFloatingPlaceholder this text will also float
+ above the input when text has been entered.
+ https://material.io/guidelines/components/text-fields.html#text-fields-input
+ */
+@property(nonatomic, nullable, copy) IBInspectable NSString *placeholder;
+
+/**
+ The color applied to the placeholder when inline (not floating.)
+
+ Default is black with Material Design hint text opacity.
+ */
+@property(nonatomic, nullable, strong) UIColor *inlinePlaceholderColor NS_SWIFT_NAME(inlineScale)
+    UI_APPEARANCE_SELECTOR;
+
+/**
+ The color applied to the placeholder when floating. However, when in error state, it will be
+ colored with the error color.
+
+ Default is black with Material Design hint text opacity.
+ */
+@property(nonatomic, nullable, strong)
+    UIColor *floatingPlaceholderColor NS_SWIFT_NAME(floatingColor) UI_APPEARANCE_SELECTOR;
+
+/**
+ The font applied to placeholder labels.
+
+ The UIFont.pointSize is respected for placeholders in an inline mode. However, the pointSize of
+ a placeholder in floating mode is placeholderFont.pointSize * floatingPlaceholderScale.
+
+ Default is system body.
+ */
+@property(nonatomic, nullable, strong) UIFont *placeholderFont UI_APPEARANCE_SELECTOR;
+
+/**
+ The scale of the the floating placeholder label in comparison to the inline placeholder specified
+ as a value from 0.0 to 1.0.
+
+ Default is 0.75.
+ */
+@property(nonatomic, assign) CGFloat floatingPlaceholderScale NS_SWIFT_NAME(floatingScale)
+    UI_APPEARANCE_SELECTOR;
+
+/** The text displayed in the text input. */
+@property(nonatomic, nullable, copy) NSString *text;
+
+/**
+ The presentation style of the text input.
+
+ Default is MDCTextInputPresentationStyleDefault.
+ */
+@property(nonatomic, assign)
+    MDCTextInputPresentationStyle presentationStyle NS_SWIFT_NAME(presentation);
+
+/**
+ The character limit for the text input. A label under the input counts characters entered and
+ presents the count / the limit.
+
+ Default is 0.
```

Comment:
Ok, I see the view mode now.  But what happens when this is 0 and the user sets a view mode?